pkg/types/ovirt/machinepool.go

```diff
@@ -1,13 +1,99 @@
 package ovirt
 
+import ovirtsdk "github.com/ovirt/go-ovirt"
+
 // MachinePool stores the configuration for a machine pool installed
 // on ovirt.
 type MachinePool struct {
+	// InstanceTypeID defines the VM instance type and overrides
+	// the hardware parameters of the created VM, including cpu and memory.
+	// If InstanceTypeID is passed, all memory and cpu variables will be ignored.
+	InstanceTypeID string `json:"instanceTypeID,omitempty"`
+
+	// CPU defines the VM CPU.
+	CPU *CPU `json:"cpu,omitempty"`
+
+	// MemoryMB is the size of a VM's memory in MiBs.
+	MemoryMB int32 `json:"memoryMB,omitempty"`
+
+	// OSDisk is the the root disk of the node.
+	OSDisk *Disk `json:"osDisk,omitempty"`
+
+	// VMType defines the workload type of the VM.
+	VMType VMType `json:"vmType,omitempty"`
+}
+
+// CPU defines the VM cpu, made of (Sockets * Cores * Threads)
+type CPU struct {
+	// Sockets is the number of sockets for a VM.
+	// Total CPUs is (Sockets * Cores * Threads)
+	Sockets int32 `json:"sockets"`
+
+	// Cores is the number of cores per socket.
+	// Total CPUs is (Sockets * Cores * Threads)
+	Cores int32 `json:"cores"`
+}
+
+// Disk defines a VM disk
+type Disk struct {
+	// SizeGB size of the bootable disk in GiB.
+	SizeGB int64 `json:"sizeGB"`
 }
 
-// Set sets the values from `required` to `a`.
-func (l *MachinePool) Set(required *MachinePool) {
-	if required == nil || l == nil {
+// VMType defines the type of the VM.
+type VMType string
+
+// Values returns a slice of all supported VMTypes.
+func (t VMType) Values() []VMType {
+	return []VMType{
+		VMTypeDesktop,
+		VMTypeServer,
+		VMTypeHighPerformance}
+}
+
+// ValidValue returns true if the value is on of the values returned by VMType.Values().
+func (t VMType) ValidValue() bool {
+	for _, v := range t.Values() {
+		if t == v {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
validation checking functions shouldn't be part of the API object, please move them to validation package.